base_layer/core/tests/support/simple_block_chain.rs

```diff
@@ -69,58 +81,48 @@ pub struct SimpleBlockChainBuilder {
-        chain.blockchain.spending_keys.push(vec![SpendInfo::new(
-            priv_key.clone(),
-            calculate_coinbase(0),
-            OutputFeatures::COINBASE_OUTPUT,
-        )]);
-        let block = BlockBuilder::new().create_coinbase(priv_key).build_with_pow();
-        chain.processes_new_block(block);
-
-        // lets mine some more blocks
-        for i in 1..(block_amount) {
-            let priv_key = PrivateKey::random(&mut rng);
-            chain.blockchain.spending_keys.push(vec![SpendInfo::new(
-                priv_key.clone(),
-                calculate_coinbase(i.into()),
-                OutputFeatures::COINBASE_OUTPUT,
-            )]);
-            let header = chain.generate_new_header();
-            let block = BlockBuilder::new()
-                .with_header(header)
-                .create_coinbase(priv_key)
-                .build_with_pow();
-            chain.processes_new_block(block);
-        }
+        // create Genesis block
+        chain.add_block(&mut rng, Vec::new());
         chain
     }
 
     /// This will add empty blocks to the chain
-    pub fn add(&mut self, block_amount: u64) {
-        let mut rng = OsRng::new().unwrap();
-        for i in 0..(block_amount) {
-            let priv_key = PrivateKey::random(&mut rng);
-            self.blockchain.spending_keys.push(vec![SpendInfo::new(
-                priv_key.clone(),
-                calculate_coinbase(i),
-                OutputFeatures::COINBASE_OUTPUT,
-            )]);
-            let header = self.generate_new_header();
-            let block = BlockBuilder::new()
-                .with_header(header)
-                .create_coinbase(priv_key)
-                .build_with_pow();
-            self.processes_new_block(block);
+    pub fn add_empty_blocks<R: Rng + CryptoRng>(&mut self, rng: &mut R, count: u64) {
+        for _ in 0..count {
+            self.add_block(rng, Vec::new())
         }
     }
 
+    /// Add a block to the chain with the given metadata
+    fn add_block<R: Rng + CryptoRng>(&mut self, rng: &mut R, tx: Vec<Transaction>) {
+        let priv_key = PrivateKey::random(rng);
+        let height = self.blockchain.blocks.len() as u64;
+        let header = if height > 0 {
+            self.generate_new_header()
+        } else {
+            self.generate_genesis_block_header()
+        };
+        let total_fee = tx
+            .iter()
+            .fold(MicroTari::default(), |tot, tx| tot + tx.get_body().get_total_fee());
+        let (cb_utxo, cb_kernel) = create_coinbase(priv_key.clone(), header.height, total_fee);
+        self.blockchain.spending_keys.push(vec![SpendInfo::new(
+            priv_key,
+            calculate_coinbase(height) + total_fee,
+            OutputFeatures::COINBASE_OUTPUT,
+        )]);
+        let block = BlockBuilder::new()
+            .with_header(header)
+            .with_coinbase_utxo(cb_utxo, cb_kernel)
+//            .with_pow(pow) // TODO
```

Comment:
Cant you add the MockPow here?
The block builder used the MockPoW with its incrementing counter here

Reply:
nvm, I got confused with code in a different branch